sdk/tests/conformance2/rendering/clear-srgb-color-buffer.html

```diff
@@ -0,0 +1,131 @@
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+<script src="../../js/glsl-conformance-test.js"></script>
+</head>
+<body>
+<canvas id="example" width="8" height="8"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+
+<script>
+"use strict";
+
+var wtu = WebGLTestUtils;
+description("This test verifies the functionality of clearing srgb color buffer.");
+
+var gl = wtu.create3DContext("example", undefined, 2);
+
+var tex = gl.createTexture();
+var fbo = gl.createFramebuffer();
+var size = 8;
+
+if (!gl) {
+    testFailed("WebGL context does not exist");
+} else {
+    testPassed("WebGL context exists");
+
+    // Create a srgb color buffer
+    init();
+
+    clear_srgb_color_buffer(0);
+    clear_srgb_color_buffer(1);
+}
+
+function checkPixel(color, expectedColor) {
+    var tolerance = 3;
+    return (Math.abs(color[0] - expectedColor[0]) <= tolerance &&
+            Math.abs(color[1] - expectedColor[1]) <= tolerance &&
+            Math.abs(color[2] - expectedColor[2]) <= tolerance &&
+            Math.abs(color[3] - expectedColor[3]) <= tolerance);
+}
+
+function init() {
+    gl.bindTexture(gl.TEXTURE_2D, tex);
+    gl.texImage2D(gl.TEXTURE_2D, 0, gl.SRGB8_ALPHA8, size, size, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+    gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, tex, 0);
+    if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) != gl.FRAMEBUFFER_COMPLETE) {
+        testFailed("Framebuffer incomplete.");
+        return;
+    } else {
+        testPassed("framebuffer complete!");
+    }
+}
+
+function clear_srgb_color_buffer(iter) {
+    debug("");
+    debug("Clear sRGB color buffer through glClear or glClearBufferfv");
+
+    var color = [0x33, 0x88, 0xbb, 0xff];
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+
+    if (iter == 0) {
+        gl.clearColor(color[0] / 255, color[1] / 255, color[2] / 255, color[3] / 255);
+        gl.clear(gl.COLOR_BUFFER_BIT);
+    } else {
+        var data = new Float32Array([color[0] / 255, color[1] / 255, color[2] / 255, color[3] / 255]);
+        gl.clearBufferfv(gl.COLOR, 0, data);
+    }
+
+    // Read pixels and compare
+    var pixels = new Uint8Array(size * size * 4);
+    gl.readPixels(0, 0, size, size, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+
+    var color_ref = wtu.linearToSRGB(color);
+    for (var ii = 0; ii < size; ++ii) {
+        for (var jj = 0; jj < size; ++jj) {
+            var index = (ii * size + jj ) * 4;
+            var pix = [pixels[index], pixels[index + 1], pixels[index + 2], pixels[index + 3]];
+            if (checkPixel(pix, color_ref) == true) {
+                testPassed("pixel at [" + jj + ", " + ii + "] is (" + pix + "). It is correct!");
+            } else {
+                testFailed("pixel at [" + jj + ", " + ii + "] should be (" + color_ref + "), but the actual color is (" + pix + ")");
```

Comment:
I'd really like to get into the habit of using WebGLUtils.checkCanvasRect instead of hand-writing these pixel comparisons into all of the tests. Could you please look into doing that refactoring?